mitodl/mit-open
# Add Posthog integration to backend

### What are the relevant tickets?

Partially closes #589 

### Description (What does it do?)

Adds support for querying feature flags via Posthog, with fallback to the standard local environment. Posthog support is toggleable and queries to Posthog are cached locally in the database in case their API becomes unavailable.

To configure:
- Set the Posthog specific `.env` file settings:
   - `POSTHOG_PROJECT_API_KEY` - the API key for the project the app should look at
   - `POSTHOG_HOST` - should be `https://app.posthog.com` generally but can be changed if needed
   - `POSTHOG_ENABLED` - boolean
- Run `./manage.py createcachetable` to create the table for the durable cache

### How can this be tested?

Automated tests should pass.

The code exposes an internal meta API at `/_/features` that provides read operations on feature flags. Listing feature flags will additionally load the cache for the default case. (Posthog identifies users via a ID for the instance and an ID for the user; the default is to use the app's hostname for the instance ID, and the environment and user ID or None for the user ID, so this is what you'll get here.) The features API is intended as a sanity-check so user-specific flag settings aren't there (but we could add that).
